app/src/test/java/com/android/streetworkapp/model/park/ParkRepositoryFirestoreTest.kt

```diff
@@ -475,4 +475,59 @@ class ParkRepositoryFirestoreTest {
     assertEquals(5, park?.occupancy)
     assertEquals(listOf("event1", "event2"), park?.events)
   }
+
+  @Test
+  fun addRatingWithNewUserUpdatesRatingSuccessfully() = runTest {
```

Comment:
Might be worth adding a test that tests the corner case when `val currentNbrRating = 1` to ensure that this case is handled if someone works on this code later on

Reply:
This test was added:
```kotlin
@Test
  fun deleteRatingWithCurrentNbrRatingOfOneResetsRatingToZero() = runTest {
    // Arrange
    val pid = "123"
    val rating = 4
    val currentRating = 4.0
    val currentNbrRating = 1

    // Mock Firestore interactions
    `when`(db.collection("parks")).thenReturn(collection)
    `when`(collection.document(pid)).thenReturn(documentRef)
    `when`(documentRef.get()).thenReturn(Tasks.forResult(document))
    `when`(document.getDouble("rating")).thenReturn(currentRating)
    `when`(document.getLong("nbrRating")).thenReturn(currentNbrRating.toLong())

    // Expected values after removing the rating
    val newNbrRating = 0
    val newRating = 0.0

    // Act
    parkRepository.deleteRating(pid, rating)

    // Assert
    verify(documentRef).update(mapOf("rating" to newRating, "nbrRating" to newNbrRating))
  }
```